discordOrigin is not defined when running npm start

The following error occurs whenever I run npm start:
ReferenceError: discordOrigin is not defined
    at userInput (C:\Users\Heather\Desktop\2bored2wait-master\main.js:427:13)
    at C:\Users\Heather\Desktop\2bored2wait-master\main.js:85:3
    at Interface._onLine (readline.js:327:5)
    at Interface._line (readline.js:658:8)
    at Interface._ttyWrite (readline.js:1003:14)
    at ReadStream.onkeypress (readline.js:205:10)
    at ReadStream.emit (events.js:315:20)
    at emitKeys (internal/readline/utils.js:335:14)
    at emitKeys.next (<anonymous>)
    at ReadStream.onData (readline.js:1137:36)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! 2bored2wait@1.0.0 start: `node main.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the 2bored2wait@1.0.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\Heather\AppData\Roaming\npm-cache\_logs\2020-09-24T15_04_25_021Z-debug.log
I made sure to change the host to 127.0.0.1 in main.js, renamed secrets.json.example to secrets.json, put in my information and discord bot token, and installed npm and git. I don't know what went wrong during this.

--- a/config.json
+++ b/config.json
@@ -17,7 +17,6 @@
         "hostname": "2b2t.org",
 	"is2b2t": true, // to allow proxies inbetween 2bored2wait and 2b2t
         "port": 25565,
-        "renderDistance": 8,
         "version": "1.12.2",
         "onlinemode": true,
         "username": "lol" // the username to use if onlinemode is false

--- a/main.js
+++ b/main.js
@@ -43,7 +43,7 @@ webserver.password = config.password
 var stoppedByPlayer = false;
 var timedStart;
 var lastQueuePlace;
-var chunkData = [];
+var chunkData = new Map();
 var loginpacket;
 let dcUser; // discord user that controlls the bot
 var totalWaitTime;
@@ -132,7 +132,10 @@ function join() {
 	client.on("packet", (data, meta) => { // each time 2b2t sends a packet
 		switch (meta.name) {
 			case "map_chunk":
-				if(config.chunkCaching) chunkData.push(data);
+				if(config.chunkCaching) chunkData.set(data.x + "_" + data.z, data);
+				break;
+			case "unload_chunk":
+				if(config.chunkCaching) chunkData.delete(data.chunkX + "_" + data.chunkZ);
 				break;
 			case "playerlist_header":
 				if (!finishedQueue && config.minecraftserver.is2b2t) { // if the packet contains the player list, we can use it to see our place in the queue
@@ -167,14 +170,14 @@ function join() {
 							finishedQueue = true;
 							webserver.queuePlace = "FINISHED";
 							webserver.ETA = "NOW";
-							activity("Queue is finished");
+							logActivity("Queue is finished");
 						}
 					}
 				}
 				break;
 			case "respawn":
 				Object.assign(loginpacket, data);
-				chunkData = [];
+				chunkData = new Map();
 				break;
 			case "login":
 				loginpacket = data;
@@ -234,21 +237,6 @@ function join() {
 		});
 
 		newProxyClient.on('packet', (data, meta) => { // redirect everything we do to 2b2t
-			let chunkPos = {};
-			if (meta.name === "position") {
-				chunkPos.x = round(data.x / 16);
-				chunkPos.z = round(data.z / 16);
-				if (chunkPos.z !== chunkPos.lx || chunkPos.x !== chunkPos.lx) {
-
-					for (let i = 0; i < chunkData.length; i++) {
-						if (chunkData[i].x < chunkPos.x - config.minecraftserver.renderDistance || chunkData[i].x > chunkPos + config.minecraftserver.renderDistance || chunkData[i].z < chunkPos.z - config.minecraftserver.renderDistance || chunkData[i] > chunkPos.z + config.minecraftserver.renderDistance) { //if a cached chunk is outside of the render distance
-							chunkData.splice(i, 1); // we delete it.
-						}
-					}
-				}
-				chunkPos.lx = chunkPos.x;
-				chunkPos.lz = chunkPos.z;
-			}
 			filterPacketAndSend(data, meta, client);
 		});
 		proxyClient = newProxyClient;
@@ -256,9 +244,9 @@ function join() {
 }
 
 function sendChunks() {
-	if(config.chunkCaching) for (let i = 0; i < chunkData.length; i++) {
-		proxyClient.write("map_chunk", chunkData[i]);
-	}
+	if(config.chunkCaching) chunkData.forEach((data) => {
+		proxyClient.write("map_chunk", data);
+	});
 }
 
 function log(logmsg) {
@@ -277,9 +265,11 @@ function log(logmsg) {
 
 function reconnect() {
 	doing = "reconnect";
-	logActivity("Reconnecting... ");
 	if (stoppedByPlayer) stoppedByPlayer = false;
-	else reconnectLoop();
+	else {
+		logActivity("Reconnecting... ");
+		reconnectLoop();
+	}
 }
 
 function reconnectLoop() {
@@ -491,7 +481,6 @@ function calcTime(msg) {
 function stopQueing() {
 	stoppedByPlayer = true;
 	stop();
-	logActivity("Queue is stopped.");
 }
 
 function logActivity(update) {

--- a/main.js
+++ b/main.js
@@ -418,7 +418,7 @@ function userInput(cmd, DiscordOrigin, discordMsg) {
 }
 
 function stopMsg(discordOrigin, discordMsg, stoppedThing) {
-	msg(discordOrigin, discordMsg && discordMsg.channel, stoppedThing, stoppedThing + " is **stopped**");
+	msg(discordOrigin, discordMsg, stoppedThing, stoppedThing + " is **stopped**");
 }
 
 function msg(discordOrigin, msg, titel, content) {

--- a/main.js
+++ b/main.js
@@ -422,7 +422,7 @@ function stopMsg(discordOrigin, discordMsg, stoppedThing) {
 }
 
 function msg(discordOrigin, msg, titel, content) {
-	if(discordOrigin) sendDicordMsg(msg.channel, titel, content);
+	if(discordOrigin) sendDiscordMsg(msg.channel, titel, content);
 	else console.log(content);
 }
 

--- a/main.js
+++ b/main.js
@@ -9,7 +9,7 @@ const discord = require('discord.js');
 const {DateTime} = require("luxon");
 const https = require("https");
 const tokens = require('prismarine-tokens-fixed');
-const save = "./saveid"
+const save = "./saveid";
 var mc_username;
 var mc_password;
 var secrets;
@@ -98,7 +98,6 @@ function stop() {
 		proxyClient.end("Stopped the proxy."); // boot the player from the server
 	}
 	server.close(); // close the server
-	activity("Queue is stopped.");
 }
 
 // function to start the whole thing
@@ -148,8 +147,7 @@ function join() {
 						ETAhour = totalWaitTime - timepassed;
 						webserver.ETA = Math.floor(ETAhour) + "h " + Math.round((ETAhour % 1) * 60) + "m";
 						server.motd = `Place in queue: ${positioninqueue} ETA: ${webserver.ETA}`; // set the MOTD because why not
-						activity("Pos: " + webserver.queuePlace + " ETA: " + webserver.ETA); //set the Discord Activity
-						log("Position in Queue: " + webserver.queuePlace)
+						logActivity("Pos: " + webserver.queuePlace + " ETA: " + webserver.ETA); //set the Discord Activity
 						if (config.notification.enabled && webserver.queuePlace <= config.notification.queuePlace && !notisend && config.discordBot && dcUser != null) {
 								sendDiscordMsg(dcUser, "Queue", "The queue is almost finished. You are in Position: " + webserver.queuePlace);
 							notisend = true;
@@ -272,13 +270,14 @@ function log(logmsg) {
 		}) + "	" + logmsg + "\n", err => {
 			if (err) console.error(err)
 		})
-		let line = rl.line;
-		process.stdout.write("\033[F\n" + logmsg + "\n$ " + line);
 	}
+	let line = rl.line;
+	process.stdout.write("\033[F\n" + logmsg + "\n$ " + line);
 }
 
 function reconnect() {
-	doing = "reconnect"
+	doing = "reconnect";
+	logActivity("Reconnecting... ");
 	if (stoppedByPlayer) stoppedByPlayer = false;
 	else reconnectLoop();
 }
@@ -428,8 +427,8 @@ function userInput(cmd, DiscordOrigin, discordMsg) {
 	}
 }
 
-function stopMsg(discordOrigin, msg, stoppedThing) {
-	msg(discordOrigin, msg.channel, stoppedThing, stoppedThing + " is **stopped**");
+function stopMsg(discordOrigin, discordMsg, stoppedThing) {
+	msg(discordOrigin, discordMsg && discordMsg.channel, stoppedThing, stoppedThing + " is **stopped**");
 }
 
 function msg(discordOrigin, msg, titel, content) {
@@ -492,6 +491,12 @@ function calcTime(msg) {
 function stopQueing() {
 	stoppedByPlayer = true;
 	stop();
+	logActivity("Queue is stopped.");
+}
+
+function logActivity(update) {
+	activity(update);
+	log(update);
 }
 module.exports = {
 	startQueue: function () {

--- a/main.js
+++ b/main.js
@@ -424,7 +424,7 @@ function userInput(cmd, DiscordOrigin, discordMsg) {
 				msg(DiscordOrigin, discordMsg, "Time calculator", "The perfect time to start the will be calculated, so you can play at " + starttimestring);
 				activity("You can play at " + starttimestring);
 			}
-			else msg(discordOrigin, discordMsg, "Error", "Unknown command");
+			else msg(DiscordOrigin, discordMsg, "Error", "Unknown command");
 	}
 }
 

